ownCloud/Client/ClientRootViewController.swift

```diff
@@ -666,6 +629,56 @@ extension ClientRootViewController : OCCoreDelegate {
 		}
 	}
 
+	func presentAuthAlert(for editBookmark: OCBookmark, error nsError: NSError?, title authFailureTitle: String, message authFailureMessage: String?, ignoreLabel authFailureIgnoreLabel: String, ignoreStyle authFailureIgnoreStyle: UIAlertAction.Style, hasEditOption authFailureHasEditOption: Bool, preferredAuthenticationMethods: [OCAuthenticationMethodIdentifier]?, completionHandler: @escaping () -> Void) {
+		let alertController = ThemedAlertController(title: authFailureTitle,
+							message: authFailureMessage,
+							preferredStyle: .alert)
+
+		alertController.addAction(UIAlertAction(title: authFailureIgnoreLabel, style: authFailureIgnoreStyle, handler: { (_) in
+			completionHandler()
+		}))
+
+		if authFailureHasEditOption {
+			alertController.addAction(UIAlertAction(title: "Sign in".localized, style: .default, handler: { [weak self] (_) in
+				completionHandler()
+
+				var notifyAuthDelegate = true
```

Comment:
@felix-schwarz reduce nesting with `guard let `statements?

Reply:
Sorry, but I didn't find any opportunity to simplify the logic there with `guard let` statements. Still could reduce nesting by unifying `if`s that were nested unnecessarily.